external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala

```diff
@@ -59,14 +59,15 @@ class AvroOptions(
    * If the option is not set, the Hadoop's config `avro.mapred.ignore.inputs.without.extension`
    * is taken into account. If the former one is not set too, file extensions are ignored.
    */
+  @deprecated("Use the general data source option pathGlobFilter for filtering file names", "3.0")
```

Comment:
I am wondering whom is this deprecation warning to? Spark users don't use `ignoreExtension` directly. I do think we should print a warning when we read & detect that `AvroFileFormat.IgnoreFilesWithoutExtensionProperty` and/or `AvroOptions.ignoreExtensionKey` are set otherwise users will never see the deprecation.

Reply:
I think it already shows an warning at https://github.com/apache/spark/pull/24518/files/d8f8420d9d3c97f96c1e09855e008ece3f275ad3#diff-8b28467c7f7a28d7fcf208a613a373c8R61

Reply:
only in one case at schema inferring. I would remove this annotation and print warning in initialization of `AvroOptions`. The deprecation warning is printed only while Spark compilation which is useless for users.

Reply:
I think we should remove `deprecated`. 

It would be great if we can put that logic into `AvroOptions` e.g.:

```scala
    parameters
      .get(AvroOptions.ignoreExtensionKey)
      .map { v =>
        logWarning(...)
        v.toBoolean
      }.getOrElse(!ignoreFilesWithoutExtension)
```

However, can you make it doesn't show the logs too many times? If we put there, seems like it will show the same logs multiple times.

Reply:
If you can find a better way, please go and open a PR (and some nits I picked below)